### PR TITLE
Updates following the removal of several np.ndarray methods.

### PR DESCRIPTION
**Pull request by mhvk**
### Description

This pull request is to address updates to numpy for 2.0, which removed the `.ptp()`, `.newbyteorder()` and `.itemset()` methods, causing `-dev` failures in current main.

Of these three methods, the latter two were basically not used, but the former is a little annoying, especially as we support it for `Time` as well. (Likely should be deprecated for `Time`, but keeping that as a separate issue [to be raised]).

I've simply removed the methods in `Quantity`, etc., as well; there are tests that the `np.ptp` function still works (adjusted for `Masked`).

Note I'm not backporting, as I think we already decided that we would not try to support numpy 2.0 for 5.3.

p.s. Review by commit may be handiest, as each commit deals with a different deprecated method.

- [X] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by mhvk** [comment]
O, darn, there was an actual bug in `Masked` - the fix in #15380 should be merged first, and then I'll rebase.

File: astropy/units/tests/test_quantity_array_methods.py [hunk 1]
```diff
@@ -468,13 +481,6 @@ def test_item(self):
         with pytest.raises(ValueError):
             q1.itemset()
 
-        q1[1] = 1
```
Comment (pllim) [inline astropy/units/tests/test_quantity_array_methods.py:471]: This removal seems unrelated to me. Why is this deleted?

File: docs/changes/units/15378.api.rst
Comment (pllim) [inline docs/changes/units/15378.api.rst:1]: If I understand the diff correctly, this entry is only relevant if someone is using numpy 2.0+?

Or are you saying that there are backward-incompatible changes even if someone is using numpy 1.x?

**Comment by pllim** [comment]
#15380 is merged. Please rebase and resolve conflict. Thanks!

File: astropy/units/tests/test_quantity_array_methods.py (hunk 1 above)
Comment (mhvk) [reply astropy/units/tests/test_quantity_array_methods.py:471]: It is still there at l.464 - basically, I split the test in two parts, one that just checks `.item()` and one that tests `.itemset` - the git diff is really confusing here!

**Comment by mhvk** [comment]
OK, rebased. I realized that I should check that also for logarithmic quantities `np.ptp(magnitude)` continues to work - and of course that meant actually adding code to make that happen. That addition is right now only to get things to stay the same for numpy 2.0 - I'll have to do a follow-up PR to make it a bit prettier/complete...

**Comment by mhvk** [comment]
See #15384 for why I'm not trying to make everything work with `__array_function__` here...

File: docs/changes/units/15378.api.rst
Comment (mhvk) [reply docs/changes/units/15378.api.rst:1]: Yes, it is only relevant for people using numpy 2.0

Comment (pllim) [reply docs/changes/units/15378.api.rst:1]: @mhvk , can you please clarify my confusion here even though diff marks it as "outdated" for some reason? Thanks!

Comment (mhvk) [reply docs/changes/units/15378.api.rst:1]: Yes, don't know why it is marked as outdated, since I didn't touch the file. Anyway, your impression is correct: it affects people using numpy 2.0. Any suggestions for making that clearer?

File: astropy/units/tests/test_quantity_array_methods.py (hunk 1 above)
Comment (pllim) [reply astropy/units/tests/test_quantity_array_methods.py:471]: Ah, I see now. Thanks for the clarification!

**Review by pllim — APPROVED** [review]
LGTM. Thanks for the quick fixes!

File: docs/changes/units/15378.api.rst
Comment (pllim) [reply docs/changes/units/15378.api.rst:1]: Now I that read it several times, I think it is fine as-is.
